### PR TITLE
Refactor unit tests without depending on Guava

### PR DESCRIPTION
**Pull request by kongleong86**
Refactor unit tests without depending on Guava external library

- Using less libraries to do unit tests will help on code maintainability and future libraries upgrade.

> For detailed contributing instructions see https://github.com/iluwatar/java-design-patterns/wiki/01.-How-to-contribute

### CODE REVIEW

File: repository/src/test/java/com/iluwatar/repository/AnnotationBasedRepositoryTest.java [hunk 1]
```diff
@@ -66,7 +65,7 @@ void setup() {
 
   @Test
   void testFindAll() {
-    var actuals = Lists.newArrayList(repository.findAll());
+    List<Person> actuals = (List<Person>) repository.findAll();
```
Comment (robertvolkmann) [inline repository/src/test/java/com/iluwatar/repository/AnnotationBasedRepositoryTest.java:68]:
```suggestion
    var actuals = (List<Person>) repository.findAll();
```

File: repository/src/test/java/com/iluwatar/repository/RepositoryTest.java [hunk 2]
```diff
@@ -66,7 +65,7 @@ void setup() {
 
   @Test
   void testFindAll() {
-    var actuals = Lists.newArrayList(repository.findAll());
+    List<Person> actuals = (List<Person>) repository.findAll();
```
Comment (robertvolkmann) [inline repository/src/test/java/com/iluwatar/repository/RepositoryTest.java:68]:
```suggestion
    var actuals = (List<Person>) repository.findAll();
```

Comment (robertvolkmann) [reply repository/src/test/java/com/iluwatar/repository/RepositoryTest.java:68]: You could add the method `List<Person> findAll` to `PersonRepository` and remove these casts.

Comment (kongleong86) [reply repository/src/test/java/com/iluwatar/repository/RepositoryTest.java]: Made the changes according to comment.

**Review by robertvolkmann — DISMISSED** [review]

**Comment by robertvolkmann** [comment]
@kongleong86 Your branch has conflicts. Could you resolve them?

File: repository/src/test/java/com/iluwatar/repository/AnnotationBasedRepositoryTest.java [hunk 3]
```diff
@@ -66,7 +65,7 @@ void setup() {
 
   @Test
   void testFindAll() {
-    var actuals = Lists.newArrayList(repository.findAll());
+    List<Person> actuals = repository.findAll();
```
Comment (robertvolkmann) [inline repository/src/test/java/com/iluwatar/repository/AnnotationBasedRepositoryTest.java:68]:
```suggestion
    var actuals = repository.findAll();
```

File: repository/src/test/java/com/iluwatar/repository/RepositoryTest.java [hunk 4]
```diff
@@ -66,7 +65,7 @@ void setup() {
 
   @Test
   void testFindAll() {
-    var actuals = Lists.newArrayList(repository.findAll());
+    List<Person> actuals = repository.findAll();
```
Comment (robertvolkmann) [inline repository/src/test/java/com/iluwatar/repository/RepositoryTest.java:68]:
```suggestion
    var actuals = repository.findAll();
```

**Review by iluwatar — CHANGES_REQUESTED** [review]
Please resolve the conflicts and we can proceed with the merge

**Comment by kongleong86** [comment]
> Please resolve the conflicts and we can proceed with the merge

Conflicts resolved.

**Review by robertvolkmann — APPROVED** [review]

**Comment by robertvolkmann** [comment]
@iluwatar could you please merge it? It would allow me to simplify #2357.

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Looks good! Thank you for the contribution 🎉

@all-contributors please add @kongleong86 for code

**Comment by iluwatar** [comment]
@all-contributors please add @robertvolkmann for review

**Pull request merged**